data/state/accountsDB.go

```diff
@@ -159,24 +200,26 @@ func (adb *AccountsDB) SaveDataTrie(accountHandler AccountHandler) error {
 		return nil
 	}
 
-	//append a journal entry as the data needs to be updated in its trie
-	entry, err := NewBaseJournalEntryData(accountHandler, accountHandler.DataTrie())
+	root, err := newDataTrie.Root()
 	if err != nil {
 		return err
 	}
 
-	adb.Journalize(entry)
-	root, err := accountHandler.DataTrie().Root()
+	err = accountHandler.SetRootHashWithJournal(root)
 	if err != nil {
 		return err
 	}
 
-	err = accountHandler.SetRootHashWithJournal(root)
+	trackableDataTrie.ClearDataCaches()
+
+	//append a journal entry as the data needs to be updated in its trie
+	entry, err := NewBaseJournalEntryData(accountHandler, newDataTrie)
```

Comment:
can you move this into a new function, that same as other saveStuffwithJournal ?

Reply:
done. saveDataTrieWithJournal